### PR TITLE
Fix affine_map construction in UNROLL_LOOPS=0 config; add e2e test cases

### LINKED ISSUE #3068: Fix inductor generation of affine_maps for loops
### What happened?

Changes to the coarse-tiling tile offset calculations have caused the non-default affine_map path to regress. 

We need to restore parity between the unrolled and non-unrolled path and then add test cases to prevent future regression. 

### What did you expect to happen?

Affine_map generation in the front-end should not have regressed. 

### How can we reproduce it?

Run simple loop unrolling example like
```
import torch

from torch_spyre._inductor import spyre_hint
import torch_spyre._inductor.propagate_named_dims as _pnd

_declare_tensor_dim = _pnd.declare_tensor_dim
_name_tensor_dims = _pnd.name_tensor_dims

A, B = 1024, 4096
a = torch.randn(A, B, dtype=torch.float16)
b = torch.randn(A, B, dtype=torch.float16)
c = torch.randn(A, B, dtype=torch.float16)


@torch.compile
def fn(a, b, c):
    # with spyre_hint(num_tiles_per_dim={"A": 2}):
    with spyre_hint(num_tiles_per_dim={"B": 4}):
        y = a + b
        z = y * c
        return z


a_dev = a.to("spyre")
b_dev = b.to("spyre")
c_dev = c.to("spyre")
_declare_tensor_dim("A", A)
_declare_tensor_dim("B", B)
_name_tensor_dims(a_dev, ["A", "B"])
_name_tensor_dims(b_dev, ["A", "B"])
_name_tensor_dims(c_dev, ["A", "B"])

cpu_res = fn(a, b, c)
spyre_res = fn(a_dev, b_dev, c_dev).cpu()

print(f"CPU result\n{cpu_res}")
print(f"Spyre Compiled result\n{spyre_res}")
cpu_delta = torch.abs(spyre_res - cpu_res).max()

print(f"Max delta Compiled Spyre vs. CPU: {cpu_delta}")
```

It produces the wrong result.  Affine_map stride in generated bundle.mlir is off by a factor of 1024.

### Any environmental details we need to know?

No,

### Anything else we need to know?

_No response_

### Relevant log output

```Shell

```

### PR DESCRIPTION
**Pull request by dgrove-oss**
#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

The `UNROLL_LOOPS=0` path emits `scf.for` loops with `affine.apply` ops in
`bundle.mlir` rather than unrolling tiles into flat `OpSpec` copies.  This PR
fixes a bug in that path that inflated the `affine.apply` stride values by
factors of up to 512×, producing wrong tile addresses and therefore wrong
results on hardware.

**Root cause** (`compute_ops.py`, `_tiled_byte_stride`):

`SDSCArgs.strides[sym]` is the per-tile element stride — it is computed by
`_calculate_device_stride` in `_create_sdsc_tensors` and further scaled by the
`dev_dim_size > it_dim_size` backGap adjustment so it already encodes the full
per-iteration tile advance.  The old implementation multiplied this value by
`per_iter_range` a second time, producing e.g. `512 × 65536 = 33554432` bytes
instead of the correct `65536` bytes for a 512-row tile.

The fix is a one-liner: drop `per_iter_range` and return
`tensor.strides[tiled_sym] * num_bytes(tensor.data_format)` directly.  The
unused `iteration_space` parameter is also removed.

**Verification** (`test_coarse_tiling.py`, `TestAffineStrideMatchesUnrollStride`):

A new test class pins the UNROLL=0 `compile_op_spec` affine strides against
the UNROLL=1 `_byte_stride_for_arg` ground truth for four cases:

| Case | Tensor | Expected stride |
|---|---|---|
| Row-tiling | [512, 256] fp16 | 512 × 64 × 2 = 65 536 B |
| Col-tiling | [512, 256] fp16 | 2 × 32 768 × 2 = 131 072 B |
| K-tiling | device_size=[2,64,64] fp16 | 2 × 4 096 × 2 = 16 384 B |
| Multi-stick row-tiling | [1024, 4096] fp16 | 512 × 64 × 2 = 65 536 B |

All seven tests were failing before the fix and pass after it.
`TestTiledByteStride` and `TestGenerateSdscTiledSymbols` are updated to
reflect the corrected semantics.

**E2E coverage** (`test_coarse_tile_e2e.py`):

With the stride fix in place, 32 of 39 e2e tests pass under `UNROLL_LOOPS=0`.
The 7 that still give wrong results are given conditional `pytest.xfail` guards
(`if not config.unroll_loops: pytest.xfail(...)`) so they do not show as
unexpected failures.  The failing tests divide into:

- **4 single-level** — `amin` dim=-1 reduction and all three dim=0 reductions:
  wrong output from the backend `scf.for` implementation for these ops.
- **3 multi-level / nested** — flash attention, nested bmm, nested matmul:
  backend `scf.for` nested loops not yet producing correct results.

Five `*Unroll0` subclasses (one per test class) are added, each decorated with
`@config.patch({"unroll_loops": False})`.  Pytest collects both the base class
(UNROLL=1) and the subclass (UNROLL=0) in a normal CI run, giving coverage of
both paths without any extra invocation:

```
39 passed           (base classes, unroll_loops=True)
32 passed, 7 xfailed (Unroll0 subclasses, unroll_loops=False)
─────────────────────────────────────────────────────────
67 passed, 7 xfailed  total
```

#### Which issue(s) this PR is related to:

Fixes #3068. 

#### Special notes for your reviewer:

- The `@config.patch` decorator on a `unittest.TestCase` *class* uses
  `setUpClass`/`tearDownClass` to apply the patch for the whole class (see
  `torch.utils._config_module.ContextDecorator.__call__`).  This is why the
  five `*Unroll0` subclasses work without any per-method change.
- Method-level `@config.patch({"unroll_loops": True})` decorators on
  `test_hint_unrolled_source_calls_sdsc` and
  `test_hint_unrolled_softmax_shaped_execution` override the class-level patch,
  so those tests continue to exercise the UNROLL=1 path even inside the
  `*Unroll0` subclass — which is intentional.
- The `_make_sdsc_spec` helper in `test_coarse_tiling.py` uses hand-crafted
  `SDSCArgs.strides` values.  After this fix, `device_stride=128` in that
  helper means "128 elements per full tile advance" (the per-tile stride), not
  "128 elements per unit step".  The updated test assertions (`128 * 2 = 256`)
  reflect this interpretation.

#### Does this PR introduce a user-facing change?

Yes — programs compiled with `UNROLL_LOOPS=0` (the `scf.for` path) previously
produced wrong output on 32 of 39 e2e coarse-tiling test cases due to
incorrect tile addresses in the emitted `affine.apply` ops.  After this fix, 32
of those 39 produce correct output; the remaining 7 are under investigation
to determine if they are inductor bugs or backend issues. 

#### Additional note:

The underlying formula error in `_tiled_byte_stride` was present since the
function was introduced.  It was invisible under `UNROLL_LOOPS=1` because that
path computes strides via `_byte_stride_for_arg` in `unroll.py` (which
evaluates device coordinates symbolically) and never calls
`_tiled_byte_stride`.

### CODE REVIEW

**Review by avery-blanchard — APPROVED** [review]
looks good to me
